### DISCUSSION TITLE
micropython hid host?

### DISCUSSION THREAD

**Discussion by insertokname** [micropython/micropython#13030] [RP2040 / Pico]
is there any possible way to make a hid host with micropython?

my current setup:
i have a raspberry pi pico w that has a female usb A connector soldered to it.
d+ is soldered to gpio0 and d- is soldered to gpio1.

I was thinking that there may be a way to use this as a usb A host.

**Comment by andrewleech** [comment]
On most chips it's not possible to add additional USB ports as it requires hardware peripheral support in the chip, but it turns out the PIO in the rp2040 _can_ handle this!
There's a few examples of people trying this out: https://hackaday.com/2022/12/28/usb-host-on-rp2040-with-pio/
That's based on the work done by https://github.com/sekigon-gonnoc/Pico-PIO-USB which is only for Arduino, I don't know of anyone trying to port that to micropython.

**Reply by insertokname** [reply]
Thank you for your reply!

I'll look into the arduino code!
